src/inspect_ai/tool/_tools/_computer/_computer_common.py

```diff
@@ -0,0 +1,127 @@
+import asyncio
+import json
+import logging
+
+from pydantic import BaseModel, Field
+
+from inspect_ai._util.content import ContentText
+from inspect_ai.model import Content, ContentImage
+from inspect_ai.tool._tool import ToolError
+from inspect_ai.tool._tools._computer._mock_logger import MockLogger
+from inspect_ai.util import sandbox
+
+# log = logging.getLogger(__name__)
+log = MockLogger()
+log.setLevel(logging.DEBUG)
+
+
+# TODO: Export ToolResult from inspect_ai.tool
+ToolResult = str | int | float | bool | list[Content]
+
+
+class ShellExecSuccessResult(BaseModel):
+    output: str | None = Field(default=None)
+    base64_image: str | None = Field(default=None)
+
+
+hackIsFirstCommand = True
+
+
+async def _send_cmd(cmdTail: list[str]) -> ToolResult:
+    # TODO: Resolve this issue
+    # without this delay, the first attempt to take a screenshot
+    # happens too soon before the GUI has actually rendered.
+    global hackIsFirstCommand
+    if hackIsFirstCommand:
+        stallResult = await sandbox().exec(["ls"])
+        if not stallResult.success:
+            log.error(f"First sandbox().exec() failed with: {stallResult.stderr}")
+            raise ToolError(f"Error executing command: {stallResult.stderr}")
+        log.debug("First sandbox().exec() succeeded...sleeping")
+        await asyncio.sleep(20)
+        log.debug("Stall done")
+        hackIsFirstCommand = False
+
+    cmd = ["python", "computer_tool_support/cli.py", "--action"] + cmdTail
+    log.debug(f"Executing command: {cmd}")
+
+    try:
+        raw_exec_result = await sandbox().exec(cmd)
+
+        if not raw_exec_result.success:
+            log.error(f"Execution failed with: {raw_exec_result.stderr[:50]}...")
+            raise ToolError(f"Error executing command: ${cmd} {raw_exec_result.stderr}")
```

Comment:
I wonder if this counts as a hard error (as we don't expect this to ever fail?). In that case we'd want to raise `RuntimeError` (let's discuss realtime)

Reply:
Let's discuss. Any non-zero exit code from the remote CLI will land here. For example, if the model sends an unsupported key name, we'll land here.

Reply:
Unsupported key name is definitely more in the realm of tool error, yes, let's discuss.